### PR TITLE
Merge develop into main

### PR DESCRIPTION
**Pull request by jard-111**

### CODE REVIEW

**Review by lucas-mrq — APPROVED** [review]

**Review by NiskuT — APPROVED** [review]

File: src/shared/Card.cpp [hunk 1]
```diff
@@ -0,0 +1,50 @@
+#include <shared.hpp>
+
+using namespace shared;
+Card::Card(CardsEnum type, unsigned difficulty, unsigned numberOfBox)
+{
+    this->type = type;
+    this->level = 1;
+    this->dificulty = difficulty;
```
Comment (jard-111) [inline src/shared/Card.cpp:8]: dificulty -> difficulty

File: src/shared/Player.cpp [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+    }
+}
+
+unsigned Player::getDificultyOfCard(CardsEnum cardType)
+{
+    for (auto card : this->listOfPriorityCard)
+    {
+        if (card->getType() == cardType)
+        {
+            return card->getDificulty();
```
Comment (jard-111) [inline src/shared/Player.cpp:243]: dificulty -> difficulty

File: src/shared/Card.cpp [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+unsigned Card::getDificulty()
+{
+    return this->dificulty;
+}
+
+void Card::setDificulty(unsigned newDificulty)
+{
+    this->dificulty = newDificulty;
+}
```
Comment (jard-111) [inline src/shared/Card.cpp:50]: dificulty -> difficulty

File: test/shared/test_card.cpp [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    shared::Card card(shared::CardsEnum::science,0,0);
+
+    BOOST_CHECK(card.getNumberOfBox() == 0);
+    card.addBox(2);
+    BOOST_CHECK(card.getNumberOfBox() == 2);
+    card.deleteBox(1);
+    BOOST_CHECK(card.getNumberOfBox() == 1);
+
+    card.setDificulty(3);
+    BOOST_CHECK(card.getDificulty() == 3);
```
Comment (jard-111) [inline test/shared/test_card.cpp:31]: dificulty -> difficulty

**Review by Vio1625 — APPROVED** [review]

**Pull request merged**